File: python/algorithms/binary_search.py
from typing import List


def binary_search(arr: List[int], target: int) -> int:
    left = 0
    right = len(arr) - 1
    while left <= right:
        mid = right + left // 2
        curr = arr[mid]
        if curr == target:
            return mid
        if curr < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1


if __name__ == "__main__":
    arr = [int(x) for x in input().split()]
    target = int(input())
    res = binary_search(arr, target)
    print(res)

File: algo/src/binary_search.go
package main

type Pair struct {
	a, b int
}

func binarySearch(arr []int, target int) Pair {
	low, high := 0, len(arr)-1

	for low <= high {
		mid := low + (high-low)/2
		if arr[mid] == target {
			return  Pair{a: mid, b: arr[mid]}
		} else if arr[mid] < target {
			low = mid + 1
		} else {
			high = mid - 1
		}
	}
	return Pair{a: -1, b: -1}
}
